### PR TITLE
Diffs - Show edit history when menu item tapped

### PR DESCRIPTION
**Pull request by staykids**
**Phabricator:** https://phabricator.wikimedia.org/T335579 (partially)

### Notes
Show the full edit history view when tapped from the toolbar menu item. I copied the `pageURL` construction methodology from the recent lead image URL fetching work here – if there's a more accepted way of constructing this URL I'd be happy to change this.

### Test Steps
1. Open a pair of diffs (either by way of page history, or from mobile web to app)
2. Tap more > article edit history in the toolbar.
3. Confirm the full article page history opens.
4. Confirm paging works correctly when scrolling this list of diffs.

### CODE REVIEW

**Review by tonisevener — COMMENTED** [review]
Commenting with a couple of thoughts, but I can merge as-is if you want to address it later.

File: Wikipedia/Code/DiffContainerViewController.swift [hunk 1]
```diff
@@ -263,7 +263,24 @@ class DiffContainerViewController: ViewController {
...
 private extension DiffContainerViewController {
-    
+
+    func fetchPageURL() -> URL? {
+        guard let articleTitle = articleTitle, let languageCode = siteURL.wmf_languageCode else {
+            return nil
+        }
+        
+        let namespaceAndTitle = articleTitle.namespaceAndTitleOfWikiResourcePath(with: languageCode)
+        guard namespaceAndTitle.namespace == .main else {
```
[the remark this answers is not in the document]
Comment (staykids) [reply Wikipedia/Code/DiffContainerViewController.swift]: Updated!

File: Wikipedia/Code/DiffContainerViewController.swift [hunk 2]
```diff
@@ -1123,7 +1140,12 @@ extension DiffContainerViewController: DiffToolbarViewDelegate {
     }
 
     func tappedEditHistory() {
-        // DIFFTODO: Push PageHistoryViewController
+        guard let articleTitle = articleTitle, let pageURL = fetchPageURL() else {
+            return
+        }
+
+        let historyViewController = PageHistoryViewController(pageTitle: articleTitle, pageURL: pageURL, articleSummaryController: diffController.articleSummaryController)
```
[the remark this answers is not in the document]
Comment (staykids) [reply Wikipedia/Code/DiffContainerViewController.swift]: @tonisevener Great catch, thank you - updated! Related to article talk history, I found a bug with paging (that also seems to occur in our current shipping version): https://phabricator.wikimedia.org/T341897

[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/DiffContainerViewController.swift]: @staykids good find!

**Review by tonisevener — APPROVED** [review]

**Pull request merged**